plugins/experimental/ts_lua/ts_lua_client_request.c

```diff
@@ -332,6 +348,10 @@ ts_lua_client_request_get_pristine_url(lua_State *L)
   ts_lua_http_ctx *http_ctx;
 
   http_ctx = ts_lua_get_http_ctx(L);
+  if (http_ctx == NULL) {
```

Comment:
given that this pattern is literally repeated everywhere, perhaps a macro is in order:
```
#define GET_CONTEXT(ctx, list) \
  ctx = ts_lua_get_http_ctx(list); \
  if (ctx == NULL) { \
   TS_ERROR(...); \
   return; \
  }
```